core/arch/arm/plat-synquacer/main.c

```diff
@@ -38,15 +39,37 @@ static struct pl011_data console_data;
 
 register_phys_mem(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE, CORE_MMU_DEVICE_SIZE);
 register_phys_mem(MEM_AREA_IO_SEC, GIC_BASE, CORE_MMU_DEVICE_SIZE);
+register_phys_mem(MEM_AREA_IO_SEC, THERMAL_SENSOR_BASE, CORE_MMU_DEVICE_SIZE);
 
 const struct thread_handlers *generic_boot_get_handlers(void)
 {
 	return &handlers;
 }
 
+static inline uint64_t read_icc_hppir1_el1(void)
+{
+	uint64_t v;
+
+	asm volatile ("mrs %0, S3_0_c12_c12_2" : "=r" (v));
+
+	return v;
+}
+
 static void main_fiq(void)
 {
-	panic();
+	uint32_t hppir1;
+
+	hppir1 = read_icc_hppir1_el1();
```

Comment:
Since you're not acking reception of the interrupt I guess we're going to receive quite a few secure interrupts.
We do have a very primitive framework for handling interrupt, there's an example of it's usage in https://github.com/OP-TEE/optee_os/blob/439203cb04cee677a6cd8bf9cb02b82d626713cc/core/arch/arm/plat-vexpress/main.c#L122-L149
and
https://github.com/OP-TEE/optee_os/blob/439203cb04cee677a6cd8bf9cb02b82d626713cc/core/arch/arm/plat-vexpress/main.c#L110-L113

If this doesn't work here perhaps we can extend it?

Reply:
Agree, I did explored this framework. It doesn't seem to handle G1 secure interrupts. Will try to extend this framework.